File: basic array ops.js
// baked using JS Rhino missing semicols

function insertinarray(a, pos, val){
    a.push(0)
    for(let i = a.length - 1; i > pos; i--)
        a[i] = a[i-1]
    a[pos] = val
}

function deletefromarray(a, pos){
    for(let i = pos; i < a.length - 1; i++)
        a[i] = a[i+1]
    a.pop()
}

function linearsearch(a, val){
    for(let i = 0; i < a.length; i++)
        if(a[i] == val)
            return true
    return false
}

let a = [1, 2, 3, 4]
insertinarray(a, 2, 10)
print(a)
deletefromarray(a, 1)
print(a)
if(linearsearch(a, 4))
    print('Found')
else
    print('Not Found')

File: basic array ops.py
def insertinarray(l, pos, val):
    l.append(0)
    for i in range(len(l)-1, pos, -1):
        l[i] = l[i-1]
    l[pos] = val

def deletefromarray(l, pos):
    for i in range(pos, len(l)-1):
        l[i] = l[i+1]
    l.pop()

def linearsearch(l, val):
    for i in range( len(l) - 1):
        if val == l[i]:
            return True
    return False
    
l = [1, 2, 3, 4]
insertinarray(l, int(input().strip()), int(input().strip()))
print(l)
deletefromarray(l, int(input().strip()))
print(l)
if(linearsearch(l, 2)):
    print("Found")
else:
    print("Not Found")
